packages/thumbprint-codemods/src/avatar-sizes/test.js

```diff
@@ -0,0 +1,167 @@
+/* eslint-disable no-console */
+const jscodeshift = require('jscodeshift');
+const codemod = require('./index');
+
+const transform = source =>
+    codemod(
+        {
+            source,
+            path: __filename,
+        },
+        {
+            jscodeshift,
+            stats: () => {},
+        },
+    );
+
+describe('output should change', () => {
+    test('simple avatars', () => {
+        expect(
+            transform(`
+            import React from 'react';
+            import { Avatar } from '@thumbtack/thumbprint-react';
+            const App = (props) => (
+                <>
+                    <Avatar size="large" />
+                    <Avatar size="medium" />
+                    <Avatar size="small" />
+                </>
+            )`),
+        ).toBe(`
+            import React from 'react';
+            import { Avatar } from '@thumbtack/thumbprint-react';
+            const App = (props) => (
+                <>
+                    <Avatar size="medium" />
+                    <Avatar size="small" />
+                    <Avatar size="xsmall" />
+                </>
+            )`);
+    });
+
+    test('renamed avatars', () => {
+        expect(
+            transform(`
+            import React from 'react';
+            import { Avatar as A } from '@thumbtack/thumbprint-react';
+            const App = (props) => (
+                <>
+                    <A size="large" />
+                    <A size="medium" />
+                    <A size="small" />
+                </>
+            )`),
+        ).toBe(`
+            import React from 'react';
+            import { Avatar as A } from '@thumbtack/thumbprint-react';
+            const App = (props) => (
+                <>
+                    <A size="medium" />
+                    <A size="small" />
+                    <A size="xsmall" />
+                </>
+            )`);
+    });
+});
+
+describe('output should change but also throw a console error', () => {
```

Comment:
Another test case that I think would be good to add:

```js
test('throws error if props are spread after a size is defined', () => {
    const codeBefore = `import React from 'react';
    import { Avatar, Input } from '@thumbtack/thumbprint-react';
    const App = (props) => (
        <div>
            <Avatar size="large" {...props} />
            <Input />
        </div>
    )`;

    const codeAfter = `import React from 'react';
    import { Avatar, Input } from '@thumbtack/thumbprint-react';
    const App = (props) => (
        <div>
            <Avatar size="medium" {...props} />
            <Input />
        </div>
    )`;

    expect(transform(codeBefore)).toBe(codeAfter);
    expect(console.error).toHaveBeenCalledTimes(1);
});
```

We care if the props are spread after the size, but don't care if props are spread and the `size` is defined after the spread.

Reply:
good test. added